components/cluster/command/check.go

```diff
@@ -167,6 +167,7 @@ func checkSystemInfo(s *cliutil.SSHConnectionProps, topo *spec.Specification, op
 					inst.OS(),
 					inst.Arch(),
 					insightVer,
+					"", // use default srcPath
```

Comment:
I think the better to define an interface for the component source file. e.g:
```
type ComponentSource interface {
    FilePath() string
}

type OSArchComponent struct {
    OS string
    Arch string
    Version string
}

func (...) FilePath() string {
    return fmt.sprintf()
}

type PatchedComponent string
func (p PatchedComponent) FilePath() string {
    return string(p)
}
```

Reply:
Maybe we should go further:
- Define a `Component` interface which contains `Deploy` or other needed methods
- Implement the `Component` interface in `TiDBComponent`, `TiSparkComponent` etc. So that we don't need to check what the component it is.

Reply:
That would be a huge refactor and is already out of the scope of this PR.